### PR TITLE
stream: update stream configuration to include/exclude

### PR DESCRIPTION
**Pull request by codeboten**
This follows https://github.com/open-telemetry/opentelemetry-specification/pull/4188 to add support to exclude attribute keys from stream configuration.

Fixes https://github.com/open-telemetry/opentelemetry-configuration/issues/112

### CODE REVIEW

**Comment by codeboten** [comment]
Will mark ready for review once the spec change is merged

**Comment by codeboten** [comment]
spec has merged, this is now ready for review

File: examples/kitchen-sink.yaml [hunk 1]
```diff
@@ -262,6 +262,8 @@ meter_provider:
           included:
             - key1
             - key2
+          excluded:
```
Comment (jack-berg) [inline examples/kitchen-sink.yaml:265]: Can you add a comment for `excluded`, and describe the priority between `included` and `excluded`?

Comment (codeboten) [reply examples/kitchen-sink.yaml:266]: Done!

File: examples/kitchen-sink.yaml [hunk 2]
```diff
@@ -262,6 +262,7 @@ meter_provider:
           included:
             - key1
             - key2
+          # Configure resource attributes to be excluded, in this example attribute service.attr1.
```
Comment (jack-berg) [inline examples/kitchen-sink.yaml]:
```suggestion
          # Configure resource attributes to be excluded, in this example attribute key3.
```
This brings up a good point. It seems like this needs to be _either_ a list to include, OR a list to exclude, but not both?

Comment (codeboten) [reply examples/kitchen-sink.yaml]: Does it? I would expect this to be similar to how resource attributes can be included or excluded. The example in the kitchen sink shows an include of `service.*` and an exclude of `service.attr1`

**Review by jack-berg — APPROVED** [review]

File: examples/kitchen-sink.yaml (hunk 2 above)
Comment (jack-berg) [reply examples/kitchen-sink.yaml]: Not sure how SDKs would handle the presence of both include and exclude lists. I know java (and probably go) can handle.

**Pull request merged**